File: pipeline/export.py
from __future__ import annotations

import json
import logging
from pathlib import Path

from .db import Database

logger = logging.getLogger(__name__)


def export_site_data(db: Database, output_dir: str) -> dict[str, int]:
    out_dir = Path(output_dir)
    out_dir.mkdir(parents=True, exist_ok=True)

    videos_path = out_dir / "videos.json"
    geo_pages_path = out_dir / "geo-pages.json"

    raw_videos = db.get_all_posted_videos()
    videos: list[dict] = []

    for video in raw_videos:
        seo = db.get_seo_metadata(video["id"]) or {}
        videos.append(
            {
                "id": video["id"],
                "content_id": video["content_id"],
                "video_path": video.get("video_path"),
                "thumbnail_path": video.get("thumbnail_path"),
                "duration_seconds": video.get("duration_seconds") or 0,
                "created_at": video["created_at"],
                "hook": video.get("hook") or "",
                "script": video.get("script") or "",
                "cta": video.get("cta") or "",
                "category": video.get("category"),
                "youtube_video_id": video.get("youtube_video_id"),
                "title": video.get("title"),
                "yt_description": video.get("yt_description"),
                "instagram_media_id": video.get("instagram_media_id"),
                "seo_title": seo.get("page_title"),
                "seo_description": seo.get("meta_description"),
                "og_title": seo.get("og_title"),
                "og_description": seo.get("og_description"),
            }
        )

    geo_pages = db.get_all_geo_pages()

    videos_path.write_text(json.dumps(videos, indent=2), encoding="utf-8")
    geo_pages_path.write_text(json.dumps(geo_pages, indent=2), encoding="utf-8")

    logger.info("Exported %s videos to %s", len(videos), videos_path)
    logger.info("Exported %s geo pages to %s", len(geo_pages), geo_pages_path)

    return {"videos": len(videos), "geo_pages": len(geo_pages)}

File: web/src/lib/server/pipeline/export.ts
import fs from "node:fs";
import path from "node:path";

import { Database } from "./db";

export function exportSiteData(db: Database): { videos: number; geoPages: number } {
  const repoRoot = path.join(process.cwd(), "..");
  const outputDir = path.join(repoRoot, "web", "src", "data");
  fs.mkdirSync(outputDir, { recursive: true });

  const videosPath = path.join(outputDir, "videos.json");
  const geoPagesPath = path.join(outputDir, "geo-pages.json");

  const rawVideos = db.get_all_posted_videos();
  const videos = rawVideos.map((video: Record<string, unknown>) => {
    const videoId = typeof video.id === "number" ? video.id : 0;
    const seo = db.get_seo_metadata(videoId);
    const getString = (key: string): string =>
      key in video && typeof video[key] === "string" ? video[key] : "";
    const getNumber = (key: string): number =>
      key in video && typeof video[key] === "number" ? video[key] : 0;

    return {
      id: videoId,
      content_id: getNumber("content_id"),
      video_path: getString("video_path") || null,
      thumbnail_path: getString("thumbnail_path") || null,
      duration_seconds: getNumber("duration_seconds"),
      created_at: getString("created_at"),
      hook: getString("hook"),
      script: getString("script"),
      cta: getString("cta"),
      category: getString("category") || null,
      youtube_video_id: getString("youtube_video_id") || null,
      title: getString("title") || null,
      yt_description: getString("yt_description") || null,
      instagram_media_id: getString("instagram_media_id") || null,
      seo_title: seo?.page_title ?? null,
      seo_description: seo?.meta_description ?? null,
      og_title: seo?.og_title ?? null,
      og_description: seo?.og_description ?? null,
    };
  });

  const geoPages = db.get_all_geo_pages();

  fs.writeFileSync(videosPath, JSON.stringify(videos, null, 2), "utf-8");
  fs.writeFileSync(geoPagesPath, JSON.stringify(geoPages, null, 2), "utf-8");

  return { videos: videos.length, geoPages: geoPages.length };
}
